dask/dataframe/multi.py

```diff
@@ -808,6 +808,25 @@ def merge_asof_indexed(left, right, **kwargs):
     name = "asof-join-indexed-" + tokenize(left, right, **kwargs)
     meta = pd.merge_asof(left._meta_nonempty, right._meta_nonempty, **kwargs)
 
+    left_empty = all(map(pd.isnull, left.divisions))
+    right_empty = all(map(pd.isnull, right.divisions))
+
+    if left_empty or right_empty:
+        if (left_empty and right_empty) or left_empty:
+            # results in an empty df that looks like ``meta``
+            return from_pandas(meta.iloc[len(meta) :], npartitions=left.npartitions)
+        else:
+            # results in an df that looks like ``left`` with nulls for
+            # all ``right.columns``
+            return map_partitions(
+                pd.merge_asof,
+                left,
+                right=right,
+                left_index=True,
+                right_index=True,
+                meta=meta,
+            )
```

Comment:
I think this can be simplified to:

```python 
if left_empty:
    # results in an empty df that looks like ``meta``
    return from_pandas(meta.iloc[len(meta) :], npartitions=left.npartitions)
elif right_empty:
    ...

Reply:
Ah great point - overthought that one a little bit.